data/tools/wesnoth/wmltools3.py

```diff
@@ -286,66 +294,126 @@ def isresource(filename):
     return ext and ext[1:] in resource_extensions
 
 def parse_macroref(start, line):
-    def handle_argument():
-        nonlocal opt_arg
-        nonlocal arg
-        nonlocal optional_args
-        nonlocal args
+    def handle_argument(buffer):
```

Comment:
Would it be possible to declare both `optional_args` and `args` as `nonlocal`? It won't have any practical effect in this specific case because Python handles mutable types by reference, but it'll make more evident that you're referring to variables defined outside this function.

Reply:
Alright